### DISCUSSION TITLE
Unable to make a Websockets connection when running on Docker but able to do so when running locally.

### DISCUSSION THREAD

**Discussion by somduttasinha** [mdn/mdn-community#517] [Content]
I am using React and Typescript for my client. I am unable to create a WebSocket connection using `new WebSocket(ws:/api-gateway/8002)` when running on a Docker container, however when I run this on my local machine (using `new WebSocket(ws:/localhost/8002)`), the connection works. Furthermore, when running the project on Docker, I am able to use a test websockets client (wscat) to connect to the server from the client environment, demonstrating that the hostname `api-gateway` should be able to be recognised by the client. I have no idea how to troubleshoot this error. Any help is much appreciated!

**Comment by bsmth** [comment]
Heya 👋🏻 

Sorry you didn't get a response on this one in a long time. If you have [general coding questions](https://developer.mozilla.org/en-US/docs/MDN/Community#general_support), you can try [Stack Overflow](https://stackoverflow.com/questions/) or other forums that can help you troubleshoot. I'm going to close this one for now. Thank you :)
